### ISSUE TITLE
Uploading empty directory throws exception

### ISSUE THREAD

**Issue by ntninja** [ipfs-shipyard/py-ipfs-http-client#191]
I just noticed that code such as the following cause an exception when it shouldn't. Need to fix this and add a test:

```py
import os, ipfshttpclient
c = ipfshttpclient.connect()
os.mkdir("somenewdirectory")
c.add("somenewdirectory")
```

results in the following since `somenewdirectory` will be empty:

```Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/ipfshttpclient/http.py", line 261, in _do_raise_for_status
    response.raise_for_status()
  File "/usr/lib/python3/dist-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: http://ipfs:5001/api/v0/add?stream-channels=true&trickle=False&only-hash=False&wrap-with-directory=False&pin=True&raw-leaves=False&nocopy=False

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.7/dist-packages/ipfshttpclient/client/files.py", line 307, in add
    resp = self._client.request('/add', decoder='json', data=body, headers=headers, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/ipfshttpclient/http.py", line 50, in wrapper
    return func(self, *args, **merged)
  File "/usr/local/lib/python3.7/dist-packages/ipfshttpclient/http.py", line 370, in request
    files, headers, data, timeout=timeout)
  File "/usr/local/lib/python3.7/dist-packages/ipfshttpclient/http.py", line 292, in _request
    self._do_raise_for_status(res)
  File "/usr/local/lib/python3.7/dist-packages/ipfshttpclient/http.py", line 279, in _do_raise_for_status
    six.raise_from(exceptions.ErrorResponse(msg, error), error)
  File "<string>", line 3, in raise_from
ipfshttpclient.exceptions.ErrorResponse: expected a file argument```

**Comment by SupraSummus** [comment]
Is this bug still there? I've added very similar test case recently: https://github.com/ipfs-shipyard/py-ipfs-http-client/commit/d008914de96467c66fb67ed6da6d45067a828be2#diff-3a766f2c5e48764d5428246e3da5c2c7R374. The only difference I see is that the dir has an absolute path.

**Comment by ntninja** [comment]
Yeah it was fixed by me in 6228ec624f7b3845b00dd011237c07094b41b40e, there are even tests there that check the empty dir case.

I'll close this then! Thanks!
